posthog/models/filters/retention_filter.py

```diff
@@ -0,0 +1,85 @@
+import datetime
+from datetime import timedelta
+from typing import Any, Dict, Optional, Tuple, Union
+
+from dateutil.relativedelta import relativedelta
+from django.http import HttpRequest
+
+from posthog.constants import (
+    PERIOD,
+    RETENTION_RECURRING,
+    RETENTION_TYPE,
+    SELECTED_INTERVAL,
+    TARGET_ENTITY,
+    TOTAL_INTERVALS,
+    TREND_FILTER_TYPE_EVENTS,
+)
+from posthog.models.entity import Entity
+from posthog.models.filters.filter import Filter
+
+RETENTION_DEFAULT_INTERVALS = 11
+
+
+class RetentionFilter(Filter):
```

Comment:
We should subclass all the insights because there's generally specific filters that don't need to be lumped into the Filter class. Also, a lot of the param processing and defaults can be set by the filter so that the query building is smoother